src/DDTrace/Integrations/OpenAI/OpenAIIntegration.php

```diff
@@ -0,0 +1,1220 @@
+                ];
+
+            case 'retrieveFineTune': // public function retrieve(string $fineTuneId): RetrieveResponse
+            case 'deleteModel': // public function delete(string $model): DeleteResponse
+            case 'cancelFineTune': // public function cancel(string $fineTuneId): RetrieveResponse
+                return [
+                    'fine_tune_id' => $args[0],
+                ];
+
+            case 'createImageEdit': // public function edit(array $parameters): EditResponse
+                /** @var array $parameters */
+                $parameters = $args[0];
+                return [
+                    'file' => $parameters['image'] ?? null,
+                    'prompt' => $parameters['prompt'] ?? null,
+                    'mask' => $parameters['mask'] ?? null,
+                    'n' => $parameters['n'] ?? null,
+                    'size' => $parameters['size'] ?? null,
+                    'response_format' => $parameters['response_format'] ?? null,
+                    'user' => $parameters['user'] ?? null,
+                ];
+
+            case 'createImageVariation': // public function variation(array $parameters): VariationResponse
+                /** @var array $parameters */
+                $parameters = $args[0];
+                return [
+                    'file' => $parameters['image'] ?? null,
+                    'n' => $parameters['n'] ?? null,
+                    'size' => $parameters['size'] ?? null,
+                    'response_format' => $parameters['response_format'] ?? null,
+                    'user' => $parameters['user'] ?? null,
+                ];
+
+            case 'createTranscription': // public function transcribe(array $parameters): TranscriptionResponse
+            case 'createTranslation': // public function translate(array $parameters): TranslationResponse
+                /** @var array $parameters */
+                $parameters = $args[0];
+                return [
+                    'file' => $parameters['file'] ?? null,
+                    'model' => $parameters['model'] ?? null,
+                    'prompt' => $parameters['prompt'] ?? null,
+                    'response_format' => $parameters['response_format'] ?? null,
+                    'temperature' => $parameters['temperature'] ?? null,
+                    'language' => $parameters['user'] ?? null,
+                ];
+        }
+
+        // Remaining OpenAI methods take a single array argument $parameters
+        return $args[0];
+    }
+
+
+    public static function handleRequest(
+        SpanData $span,
+        string   $methodName,
+        array    $args,
+        string   $basePath,
+        string   $apiKey,
+        bool     $streamed = false,
+    )
+    {
+        $payload = self::normalizeRequestPayload($methodName, $args);
+
+        $span->name = 'openai.request';
+        $span->resource = isset($payload['model']) ? $methodName . '/' . $payload['model'] : $methodName;
```

Comment:
I think this doesn't match what the doc defines:

> Resource name:  the resource name shall be: `<endpoint-name>/<model-name>`. For example, using the chat/completions endpoint and the model gpt-3.5-turbo, the resource name would be: `chat.completions/gpt-3.5-turbo`. If the model cannot be retrieved for some reason, the resource name is `<endpoint-name>`.

For example, in the case of chat completions we would end up with `createChatCompletion/gpt-3.5-turbo` instead right?

Reply:
Nice catch